internal/scaffold/cmd.go

```diff
@@ -151,10 +151,29 @@ func main() {
 		os.Exit(1)
 	}
 
-	if err = serveCRMetrics(cfg); err != nil {
-		log.Info("Could not generate and serve custom resource metrics", "error", err.Error())
+	// Add the Metrics Service
+	addMetrics(cfg, ctx, namespace)
+
+	log.Info("Starting the Cmd.")
+
+	// Start the Cmd
+	if err := mgr.Start(signals.SetupSignalHandler()); err != nil {
+		log.Error(err, "Manager exited non-zero")
+		os.Exit(1)
+	}
+}
+
+// addMetrics will create the Services and Service Monitors to allow the operator export the metrics by using
+// the Prometheus operator
+func addMetrics(cfg *rest.Config, ctx context.Context, namespace string) {
+	if k8sutil.IsRunModeLocal() {
+		log.Info("Skipping service monitors creation; not running in a cluster.")
+		return
 	}
 
+	if err := serveCRMetrics(cfg); err != nil {
+		log.Info("Could not generate and serve custom resource metrics", "error", err.Error())
```

Comment:
The error returned here can still be `k8sutil.ErrRunLocal`, so check the error in `serveCRMetrics` below like in [this example](https://github.com/estroz/operator-sdk/commit/0eb6414091822531d8943d56d45d3f3f2146f88d).